README.md

```diff
@@ -1,39 +1,134 @@
+configured).
 
-To run an instance of NATS, checkout the [local-dev](https://github.com/compliance-framework/local-dev) repository: 
+To run an instance of NATS, checkout the [local-dev](https://github.com/compliance-framework/local-dev) repository:
+
+## Configuration
+
+The agent must be configured using a configuration file that can be in any of YAML, JSON or TOML. We'll assume YAML
+because it's fairly human readable and wide spread.
+
+The basic configuration file looks something like the following fields:
+
+```
+nats:
+  domain: <nats_domain>
+  port: <nats_port>
+
+plugins:
+  <plugin_identifier>:  # Can have as many of these as you like
+    assessment_plan_id: <assessment_plan_id>
+    source: <plugin_source>
+    policies:
+      - <policy1>
+      - <policy2>
+      ...
+    config:
+      <config1>: <value1>
+      <config2>: <value2>
+      ...
+```
+
+The `nats_domain` and `nats_port` items are the domain and port of the NATS server that the agent will connect to.
+
+The `plugin_identifier` is a unique identifier for the plugin, and is used to identify the plugin in the logs, you can
+name this whatever you like but it must be unique.
+
+The `assessment_plan_id` is the id of the assessment plan that the plugin is associated with.
+
+The `plugin_source` is the path to the plugin binary that the agent will run. This can be a relative or absolute path or
+even a URL to a remote plugin.
+
+The `policies` field is a list of paths to the policy files that the plugin will use to assess the data it collects.
+
+The `config` field is a map of configuration values that the plugin will use to connect to the data source. The values
+will be passed to the plugin when it is run.
+
+You can specify as many plugins as you wish, as long as each identifier is unique. You can even reuse the same plugin
+multiple times with different configurations.
+
+As an example, a configuration file might look like this:
+```yaml
+nats:
+  domain: localhost
+  port: 4222
+
+plugins:
+  local-ssh-security:
+    assessment_plan_id: "12341234-1234-1234-123412341234"
+
+    source: "../plugin-local-ssh/cf-plugin-local-ssh"
+    policies:
+      - "../plugin-local-ssh-policies/dist/bundle.tar.gz"
+
+    config:
+      host: "10.0.0.4"
+      username: "user"
+      password: "password"
+
+  local-ssh-security2:
+    assessment_plan_id: "45674567-4567-4567-456745674567"
+
+    schedule: "* * * * *"
+
+    source: "../plugin-local-ssh/cf-plugin-local-ssh"
+    policies:
+      - "../plugin-local-ssh-policies/dist/bundle.tar.gz"
+
+    config:
+      host: "10.0.0.5"
+      username: "user"
+      password: "password"
+```
+
+Note that if you run the agent as a daemon and change the configuration file it will automatically reload the
+configuration file and update the plugins and policies that are running.
+
+More documentation on the configuration of the agent can be found in the [configuration](./docs/configuration.md)
+documentation.
```

Comment:
All of this should be included in the configuration doc, or even better an example *full configuration file. 

Adding this in the README might be overwhelming to someone just trying to understand what the agent is actually about.